doc/application/application.rst

```diff
@@ -831,44 +834,48 @@ Application Configuration
 Kconfig Configuration
 =====================
 
-The application is configured using a set of options that can be customized for
-application-specific purposes. The Zephyr build system takes a configuration
-option's value from the first source in which it is specified, taken from the
-following available sources, in order:
+The configuration for an application is produced by merging configuration
```

Comment:
This is not quite true if a .config exists and it is modified. The previous text took that into account

Reply:
I actually like your rewrite, but we should warn users about that .config straight away I feel, or at least mention that what you say only applies to clean builds and builds where you have modified your .conf files

Reply:
I updated it to say "The initial configuration for an application..." Do you think that will make it clearer?

Reply:
OK, that is fine